andela/technites-bn-backend
# #168050001 Reset password

#### What does this PR do?

- This PR Contains full validations, tests for password reset and refactoring the previous approach to signing limited time token
#### Description of Task to be completed?

- Adding password complexity

- Writing tests

- Overriding the method for signing tokens

#### How should this be manually tested?

1. Clone repository `$ git clone https://github.com/andela/technites-bn-backend.git`

2. Run `git checkout reset-password-via-email-168050001` to your terminal.

3. Run `npm install` 

4. Remember to add the variable `SENDGRID_API_KEY` to your `.env` file with the corresponding value. 

5. Run `npm run dev`

6. run `sequelize db:migrate` to create PasswordResets table and models

7. make sure you already have a registered user or just run `sequelize db:seed:all `, you will be having an account with `email:technitesdev@gmail.com`, you can still edit it and assign it your own email to check it.

8. To request email reset link  use this route `/api/v1/auth/reset` using `POST` method, you will also need to add in an `email` field.

9. Check your email and copy the link to postman using `PATCH` method and pass in `password` and `confirm_password` fields.

10. You will be receiving appropriate messages if the password doesn't fill all the requirements, password needs to be having one uppercase letter, one lowercase letter


#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168050001](https://www.pivotaltracker.com/story/show/168050001)
#### What are the relevant Github Issues?
N/A
#### Screenshots (if appropriate)
<img width="1122" alt="Screen Shot 2019-09-10 at 16 36 04" src="https://user-images.githubusercontent.com/42216593/64623988-3d8d7700-d3ea-11e9-8780-677c7b071f45.png">
<img width="1118" alt="Screen Shot 2019-09-10 at 16 36 19" src="https://user-images.githubusercontent.com/42216593/64623989-3d8d7700-d3ea-11e9-888b-7687283e2d72.png">
<img width="812" alt="Screen Shot 2019-09-10 at 16 38 56" src="https://user-images.githubusercontent.com/42216593/64623992-3e260d80-d3ea-11e9-98a7-1d329423ba72.png">


#### Questions:
N/A
